openmc/mgxs/mgxs.py

```diff
@@ -3891,40 +4201,70 @@ def get_xs(self, in_groups='all', out_groups='all',
             num_mu_bins = 1
 
         # Reshape tally data array with separate axes for domain and energy
-        num_subdomains = int(xs.shape[0] /
-                             (num_mu_bins * num_in_groups * num_out_groups))
-        if self.scatter_format == 'histogram':
-            new_shape = (num_subdomains, num_in_groups, num_out_groups,
-                         num_mu_bins)
+        # Accomodate the polar and azimuthal bins if needed
+        num_subdomains = int(xs.shape[0] / (num_mu_bins * num_in_groups *
+                                            num_out_groups * self.num_polar *
+                                            self.num_azimuthal))
+        if self.num_polar > 1 or self.num_azimuthal > 1:
+            if self.scatter_format == 'histogram':
+                new_shape = (self.num_polar, self.num_azimuthal,
```

Comment:
Definitely agree with the intent, see the comment immediately above this - will hold off for now